tao_compiler/mlir/xla/ral/context/common_context_impl_mkldnn.cc

```diff
@@ -607,21 +581,117 @@ struct OnednnACLKeyHasher {
   }
 };
 
-using MatmulPrimitive = ideep::matmul_forward::super;
 template <typename TKey, typename TValue>
-using OnednnACLGemmKeyMap =
-    std::unordered_map<TKey, TValue, OnednnACLKeyHasher>;
+using CpuGemmKeyMap = std::unordered_map<TKey, TValue, CpuGemmKeyHasher>;
+
+#if defined(TAO_X86)
+class MklPackedWeight {
+ public:
+  MklPackedWeight(ExecutionContext* ctx, const CpuGemmKey& key);
+  ~MklPackedWeight();
+
+  opaque_t packed_weight() const { return packed_weight_; }
+
+ private:
+  Context* ctx_;
+  cpu::CPUDriver* driver_;
+  opaque_t packed_weight_ = nullptr;
+};
+
+MklPackedWeight::MklPackedWeight(ExecutionContext* ctx, const CpuGemmKey& key)
+    : ctx_(ctx->getContext()),
+      driver_(ctx->getDriver<cpu::CPUDriver>(cpu::CPUDriver::name())) {
+  size_t packed_weight_size =
+      cblas_sgemm_pack_get_size(CblasBMatrix, key.m, key.n, key.k);
+  packed_weight_ = driver_->raw_alloc(ctx_, packed_weight_size);
+  cblas_sgemm_pack(
```

Comment:
Does that mean the pre-pack is a strategy that is related to the shapes of the input and transpose operations?

Reply:
Yes, we need to take transpose attribute into consideration as the document of mkl shown. Here are some exampels:
https://www.intel.com/content/www/us/en/developer/articles/technical/introducing-the-new-packed-apis-for-gemm.html